src/DataTable/DataTable.jsx

```diff
@@ -14,38 +14,46 @@ export class DataTable extends Component {
   }
   
   static defaultProps = {
-    autoResize: false
+    autoResize: true
   }
-  
-  handleResize (e) {
+
+  handleResize (grid) {
     if (this.autoResize) {
-      this.refs.grid.api.sizeColumnsToFit()
+      grid.api.sizeColumnsToFit()
     }
   }
-
+  
   componentDidMount () {
-    window.addEventListener('resize', this.handleResize.bind(this))
+    const grid = this.refs.grid
+  
+    if (this.autoResize) {
+      grid.api.sizeColumnsToFit()
+    }
+  
+    this.boundResize = this.handleResize.bind(this, grid)
+  
+    window.addEventListener('resize', this.boundResize)
   }
-
+  
   componentWillUnmount () {
-    window.removeEventListener('resize', this.handleResize.bind(this))
+    window.removeEventListener('resize', this.boundResize)
```

Comment:
I would watch with binding to `resize` since it can fire hundreds or thousands of times when resizing. we should probably debounce this call.

can use this: [lodash.debounce](https://lodash.com/docs#debounce)

```javascript
import debounce from 'lodash.debounce'

// inside the class definition

componentDidMount () {
  const grid = this.refs.grid
  
  // Run initially instead of writing the if statement
  this.handleResize(grid)
  
  this.handleResize = debounce(this.handleResize.bind(this, grid), 300) // <-- I chose 300ms here, but just toy around to see what works best

  window.addEventListener('resize', this.handleResize)
}
```